src/main/java/io/burt/athena/result/s3/InputStreamResponseTransformer.java

```diff
@@ -65,24 +70,35 @@ public void exceptionOccurred(Throwable t) {
     public void onSubscribe(Subscription s) {
         subscription = s;
         if (response.contentLength() < TARGET_BUFFER_SIZE) {
+            requests.set(Integer.MAX_VALUE);
             subscription.request(Long.MAX_VALUE);
         } else {
+            requests.set(10);
             subscription.request(10L);
         }
     }
 
     @Override
     public void onNext(ByteBuffer byteBuffer) {
-        if(byteBuffer.hasRemaining()) {
+        int remaining = byteBuffer.remaining();
```

Comment:
I find the variable name `remaining` confusing since it is the whole buffer at this point. I would have called it `chunkSize`.